pkg/transportrequest/rfc/upload.go

```diff
@@ -0,0 +1,152 @@
+package rfc
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/config/validation"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+const (
+	eq = "="
+)
+
+// Exec Everything we need for calling an executable
+type Exec interface {
+	command.ExecRunner
+	GetExitCode() int
+}
+
+// Upload ...
+type Upload interface {
+	Perform(Exec) error
+	WithConnection(Connection)
+	WithConfiguration(UploadConfig)
+	WithApplication(Application)
+	WithTransportRequestID(string)
+	WithApplicationURL(string)
+}
+
+// Connection Everything we need for connecting to the ABAP system
+type Connection struct {
+	// The endpoint in for form <protocol>://<host>:<port>, no path
+	Endpoint string
+	// The ABAP client, like e.g. "001"
+	Client string
+	// The ABAP instance, like e.g. "DEV",  "QA"
+	Instance string
+	User     string
+	Password string
+}
+
+// Application Application specific properties
+type Application struct {
+	Name        string
+	Description string
+	AbapPackage string
+}
+
+// UploadConfig additional configuration properties
+type UploadConfig struct {
+	AcceptUnixStyleEndOfLine bool
+	CodePage                 string
+	FailUploadOnWarning      bool
+	Verbose                  bool
+}
+
+// UploadAction Collects all the properties we need for the deployment
+type UploadAction struct {
+	Connection         Connection
+	Application        Application
+	Configuration      UploadConfig
+	TransportRequestID string
+	ApplicationURL     string
+}
+
+// WithApplicationURL The location of the deployable
+func (action *UploadAction) WithApplicationURL(z string) {
+	action.ApplicationURL = z
+}
+
+// WithTransportRequestID The transport request ID for the upload
+func (action *UploadAction) WithTransportRequestID(t string) {
+	action.TransportRequestID = t
+}
+
+// WithApplication Everything we need to know about the application
+func (action *UploadAction) WithApplication(a Application) {
+	action.Application = a
+}
+
+// WithConfiguration Everything we need to know in order to perform the upload
+func (action *UploadAction) WithConfiguration(c UploadConfig) {
+	action.Configuration = c
+}
+
+// WithConnection Everything we need to know about the connection
+func (action *UploadAction) WithConnection(c Connection) {
+	action.Connection = c
+}
+
+// Perform Performs the upload
+func (action *UploadAction) Perform(command Exec) error {
+
+	log.Entry().Infof("Deploying artifact '%s' to '%s', client: '%s', instance: '%s'.",
+		action.ApplicationURL, action.Connection.Endpoint, action.Connection.Client, action.Connection.Instance,
+	)
+
+	missingParameters, err := validation.FindEmptyStringsInConfigStruct(*action)
+	if err != nil {
+		return fmt.Errorf("Cannot check that all required parameters are available for SOLMAN upload: %w", err)
+	}
+	notInitialized := len(missingParameters) != 0
+	if notInitialized {
```

Comment:
`if ( len(missingParameters) != 0)`  would be fine

Reply:
dropped. Now let's wait for the next review who might suggest to use a dedicated variable here with a speaking name in order to make the semantics more expclict :-)